### DISCUSSION TITLE
Save parameters into a list

### DISCUSSION THREAD

**Discussion by bruno-luca** [yamcs/yamcs#675] [Q&A] [answered]
Hi everyone!
I was following [Python Yamcs Client](https://docs.yamcs.org/python-yamcs-client/) in order to add some features to my project and i was wondering: is there a way to get a list of all available parameters after i connect to Yamcs?

What i would like something like this:
```python
from yamcs.client import YamcsClient

# Connection
client = YamcsClient('localhost:8090')
archive = client.get_archive(instance='myproject')

#Get a list of all parameters
params = archive._______()

"""
So the list contains
params = ['Voltage', 'temperature', 'memory_used', ... ]
"""

for param in params:
    #print the value of the param

```

**Comment by fqqb** [comment, marked as answer]
You can get lists of known parameters from the "Mission Database" with this method:
https://docs.yamcs.org/python-yamcs-client/mdb/client/#yamcs.mdb.client.MDBClient.list_parameters

Then to get the current value of a bunch of parameters, you can use this method:
https://docs.yamcs.org/python-yamcs-client/tmtc/client/#yamcs.tmtc.client.ProcessorClient.get_parameter_values

So something like this:

```python
mdb = client.get_mdb(instance='myproject')
processor = client.get_processor(instance='myproject', processor='realtime')

params = list(mdb.list_parameters())
pvals = processor.get_parameter_values(params)
```

**Reply by fqqb** [reply]
Actually params should be more something like this:
```python
params = [p.name for p in mdb.list_parameters()]
```
